### PR TITLE
OnTap-updates

### PR DESCRIPTION
**Pull request by sibteali786**
## Describe your changes
- added onTap to have access to local scope 
- similar for pickFiles action
## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/runtime/hooks/useEnsembleAction.tsx [hunk 1]
```diff
@@ -323,10 +324,10 @@ export const usePickFiles: EnsembleActionHook<PickFilesAction> = (
 
   useEffect(() => {
     if (!isEmpty(values?.files) && !isComplete) {
-      onCompleteAction?.callback({ files: values?.files });
+      onCompleteAction?.callback({ files: values?.files, ...scope });
```
Comment (evshi) [inline packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:327]: What action are you trying to invoke here that is missing the scope?

Comment (sibteali786) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:327]: i am trying to access the item-template `name` under `pickFiles` to modify the the data

Comment (evshi) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:327]: Can you share the YAML block? The `pickFiles` action should have access to the custom scope via `useRegisterBindings`

Comment (sibteali786) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:327]: yes here 
```yaml
Button:
                              label: Upload File
                              onTap:
                                pickFiles:
                                  id: ${"uploadDoc"+docList.index}
                                  allowMultiple: true
                                  allowedExtensions:
                                    - pdf
                                    - docx
                                  onComplete:
                                    executeCode: |
                                      const fileName = files[0].name;
                                      const date = formatDate(files[0].lastModifiedDate);
                                      const data = ensemble.storage.get('docsData');
                                      const newData = data.map((d) => {
                                        if(d.index === docList.index){
                                          d.fileName = fileName;
                                          d.date = date;
                                        }
                                        return d;
                                      });
                                      ensemble.storage.set('docsData', newData);
```

Comment (evshi) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:327]: So the ID is the issue? For executeCode, we would need to add the custom scope hook there.

Comment (sibteali786) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:327]: so are we gonna approve it @evshi ?

**Pull request merged**